app/javascript/components/supplemental_application/sections/PreferencesTable.js

```diff
@@ -1,35 +1,37 @@
 import React from 'react'
+import { map, reject, isEmpty, overSome, findIndex } from 'lodash'
 
-import { map, reject, filter, isEmpty, overSome, find } from 'lodash'
-import Icon from '~/components/atoms/Icon'
+import TableWrapper from '~/components/atoms/TableWrapper'
+import PreferenceIcon from './preferences/PreferenceIcon'
 import ExpandableTable from '~/components/molecules/ExpandableTable'
-import appPaths from '~/utils/appPaths'
 import Panel from './preferences/Panel'
 import {
   isCOP,
   isDTHP,
-  isGriffith,
+  isAliceGriffith,
   getPreferenceName
 } from './preferences/utils'
+import { getTypeOfProof } from './preferences/typeOfProof'
+import { getFullHousehold } from '~/components/applications/application_form/preferences/utils'
 
 const { ExpanderButton } = ExpandableTable
 
-const hasExpanderButton = (prefName) =>
-  !(isCOP(prefName) || isDTHP(prefName) || isGriffith(prefName))
+const hasExpanderButton = (prefName) => !overSome(isCOP, isDTHP, isAliceGriffith)(prefName)
 
-const getAttachments = (preference, proofFiles, fileBaseUrl) => {
-  const selectedProofFiles = filter(proofFiles, { related_application_preference: preference.id })
-  return (!isEmpty(selectedProofFiles) &&
-          <ProofFilesList proofFiles={selectedProofFiles} fileBaseUrl={fileBaseUrl} />)
+const onlyValid = (preferences) => {
+  return reject(preferences, (pref) => {
+    return !pref.receives_preference ||
+           pref.lottery_status === 'Invalid for lottery' ||
+           isEmpty(pref.application_member)
+  })
 }
 
-const getTypeOfProof = (preference, proofFiles, fileBaseUrl) => {
-  if (overSome(isCOP, isDTHP)(preference.preference_name))
-    return preference.certificate_number
-  else
-    return getAttachments(preference, proofFiles, fileBaseUrl)
+const matchingPreference = (row) => (preference) => {
+  return getPreferenceName(preference) === row[1].content
 }
 
+/** Presenter **/
```

Comment:
what does this mean? looks like it was here before, but still curious why it's here and what it means

Reply:
I wanted to make a distinction between the presentational code and the utilities